nextmv/nextmv/local/executor.py

```diff
@@ -0,0 +1,565 @@
+from nextmv.local.geojson_handler import handle_geojson_visual
+from nextmv.local.plotly_handler import handle_plotly_visual
+from nextmv.local.runner import calculate_files_size
+from nextmv.output import (
+    ASSETS_KEY,
+    DEFAULT_OUTPUT_JSON_FILE,
+    LOGS_FILE,
+    LOGS_KEY,
+    OUTPUTS_KEY,
+    SOLUTIONS_KEY,
+    STATISTICS_KEY,
+    Asset,
+    OutputFormat,
+    VisualSchema,
+)
+from nextmv.status import StatusV2
+
+
+def main() -> None:
+    """
+    Main function to execute a local run. This function is called when
+    executing the script directly. It loads input data (arguments) from stdin
+    and orders the execution of the run.
+    """
+
+    input = load()
+    execute_run(
+        run_id=input.data["run_id"],
+        src=input.data["src"],
+        manifest_entrypoint=input.data["manifest_entrypoint"],
+        run_dir=input.data["run_dir"],
+        run_config=input.data["run_config"],
+        inputs_dir_path=input.data["inputs_dir_path"],
+        options=input.data["options"],
+        input_data=input.data["input_data"],
+    )
+
+
+def execute_run(
+    run_id: str,
+    src: str,
+    manifest_entrypoint: str,
+    run_dir: str,
+    run_config: dict[str, Any],
+    inputs_dir_path: Optional[str] = None,
+    options: Optional[dict[str, Any]] = None,
+    input_data: Optional[Union[dict[str, Any], str]] = None,
+) -> None:
+    """
+    This function actually executes the decision model run, using a
+    subprocess to call the entrypoint script with the appropriate input and
+    options.
+
+    Parameters
+    ----------
+    src : str
+        The path to the application source code.
+    manifest_entrypoint : str
+        The entrypoint script as defined in the application manifest.
+    run_dir : str
+        The path to the run directory.
+    run_config : dict[str, Any]
+        The run configuration.
+    inputs_dir_path : Optional[str], optional
+        The path to the directory containing input files, by default None. If
+        provided, this parameter takes precedence over `input_data`.
+    options : Optional[dict[str, Any]], optional
+        Additional options for the run, by default None.
+    input_data : Optional[Union[dict[str, Any], str]], optional
+        The input data for the run, by default None. If `inputs_dir_path` is
+        provided, this parameter is ignored.
+    """
+
+    # Create the logs dir to register whatever failure might happen during the
+    # execution process.
+    logs_dir = os.path.join(run_dir, LOGS_KEY)
+    os.makedirs(logs_dir, exist_ok=True)
+
+    # The complete execution is wrapped to capture any errors.
+    try:
+        # Create a temp dir, and copy the entire src there, to have a transient
+        # place to work from, and be cleaned up afterwards.
+        with tempfile.TemporaryDirectory() as temp_dir:
+            temp_src = os.path.join(temp_dir, "src")
+            shutil.copytree(src, temp_src, ignore=shutil.ignore_patterns(".nextmv"))
```

Comment:
We are not copying / handling the requirements from the manifest / requirements.txt file, right? :thinking: 
Is this a plan in the future, or, do we simply use the local installation for the local experience?

Reply:
We use the local installation given by `sys.executable`, so we don't care much about the requirements at the moment.